source/s3_auto_ranged_put.c

```diff
@@ -1206,9 +990,15 @@ static void s_s3_prepare_upload_part_on_read_done(void *user_data) {
         --auto_ranged_put->synced_data.num_parts_pending_read;
 
         auto_ranged_put->synced_data.is_body_stream_at_end = is_body_stream_at_end;
+        if (request->uploaded_before_resume) {
```

Comment:
1) remove the `{ }` block around the critical section...
2) ... so we can move the part_info declaration close to where we set it
```suggestion
    struct aws_s3_mpu_part_info *previously_uploaded_info = NULL;
    if (request->was_previously_uploaded) {
        aws_array_list_get_at(
                &auto_ranged_put->synced_data.part_list, &previously_uploaded_info, request->part_number - 1);
```
those blocks are just to be decorative and make the critical section stand out more. But now it's making the code harder to read